doc/box2d.md

```diff
@@ -4,13 +4,59 @@ Although Flame does not implements Box2d itself, it bundles a forked port of the
 
 The source of the bundled box2d on Flame can be found [here](https://github.com/feroult/box2d.dart).
 
-There is also a simple example game that can be used as reference of how to use box2d on Flame [here](https://github.com/feroult/haunt).
+There is a simple example game [here](https://github.com/feroult/haunt) that can be used as reference of how to use box2d on Flame, although it is a little bit outdated and doesn't use `Box2DGame`.
+
+There are some updated examples of how to use it [here](./examples/box2d), but they are not full game implementations.
 
 ## BaseGame extension
 
-If you are going to use Box2D in your project it can be a good idea to use the Box2D specific
-extension of the `BaseGame` class. It is called `Box2DGame` and it will control the adding and
-removal of Box2D's BodyComponents from your root `Box2DComponent`.
+If you are going to use Box2D in your project it can be a good idea to use the Box2D specific extension of the `BaseGame` class. It is called `Box2DGame` and it will control the adding and removal of Box2D's BodyComponents from your root `Box2DComponent`.
+
+A simple `Box2DGame` implementation example can be seen in the [examples folder](./examples/box2d/simple).
+
+## Contact callbacks
+
+If you are using `Box2DGame` you can take advantage of its way of handling contacts between two `BodyComponent`s.
+
+When creating the body definition for your `BodyComponent` make sure that you set the userdata to the current object, otherwise it will not be possible to detect collisions.
+Like this:
+```dart
+final bodyDef = BodyDef()
+  // To be able to determine object in collision
+  ..setUserData(this);
+```
+
+Now you have to make an implementation of `ContactCallback` where you set which two types that it should react when they come in contact.
+If you have two `BodyComponent`s named Ball and Wall and you want to do something when they come in contact you would do like this:
+
+```dart
+class BallWallCallback implements ContactCallback<Ball, Wall> {
+  BallWallCallback();
+
+  @override
+  void begin(Ball ball, Wall wall) {
```

Comment:
you need to add the new param Contact contact here :)

Reply:
Good catch! Fixed now :)